### ISSUE TITLE
After pausing the websocket connection from Apollo iOS, it can't reconnect.

### ISSUE THREAD

**Issue by cshadek** [d-exclaimation/pioneer#110]
**Describe the bug**

So my team is using Apollo iOS on our frontend and Pioneer on the backend.  I'm noticing that when we pause the websocket connection on the frontend (i.e., when backgrounding the app) that we are unable to reconnect the websocket later on.  Perhaps this is an issue with Apollo iOS, but given that no one else is having this issue while using ApolloServer, I suspect this is a bug within Pioneer.

Front end code:
```
print(transport.isConnected()) // Prints true
transport.pauseWebSocketConnection() 
print(transport.isConnected()) // Prints false
transport.resumeWebSocketConnection(autoReconnect: true)
print(transport.isConnected()) // Prints false (should print true)
```

We're using Pioneer 1.0 with the defaults and the latest version of Apollo iOS.

**Comment by d-exclaimation** [comment]
I'm not particularly sure why that's the case, but I'll look into it.

**Comment by d-exclaimation** [comment]
It seems like pausing the websocket transport just a regular disconnect, and reconnecting just starts over the connection. I am not sure why that wouldn't work. The only thing might be a problem is if the reconnect HTTP Upgrade request is different from the initial one

**Comment by cshadek** [comment]
> It seems like pausing the websocket transport just a regular disconnect, and reconnecting just starts over the connection. I am not sure why that wouldn't work. The only thing might be a problem is if the reconnect HTTP Upgrade request is different from the initial one

@calvincestari are you able to shed light on this?

**Comment by d-exclaimation** [comment]
> It seems like pausing the websocket transport just a regular disconnect, and reconnecting just starts over the connection. I am not sure why that wouldn't work. The only thing might be a problem is if the reconnect HTTP Upgrade request is different from the initial one

I will try recreating the issue with a custom test server so I can see what the server is receiving hopefully before the end of the week.

**Comment by calvincestari** [comment]
> @calvincestari are you able to shed light on this?

Looking through that piece of code I don't see anything that would make it different.

> I will try recreating the issue with a custom test server so I can see what the server is receiving hopefully before the end of the week.

I'm not familiar with Pioneer but if you get that test server up I can take a look from our end what the issue could be.

**Comment by calvincestari** [comment]
@cshadek you mentioned in the issue on our repo that you're updating the connecting payload between pause and reconnection - could that be the issue? What happens if you try without updating the payload?

https://github.com/apollographql/apollo-ios/issues/2675#issuecomment-1326857765

**Comment by cshadek** [comment]
> @cshadek you mentioned in the issue on our repo that you're updating the connecting payload between pause and reconnection - could that be the issue? What happens if you try without updating the payload?
> 
> 
> 
> https://github.com/apollographql/apollo-ios/issues/2675#issuecomment-1326857765

Thanks @calvincestari. In this case I'm actually not updating the payload.  I'm calling pause when the app leaves the foreground and then calling resume when the app re-enters the foreground.

I've also just tried calling pause and then calling resume sequentially to test it, and I get the same result.  The web socket doesn't resume the connection.

**Comment by cshadek** [comment]
So I did some more investigation into this and found that resuming the websocket isn't instantaneous.  See here:

https://github.com/apollographql/apollo-ios/issues/2675#issuecomment-1334082625

**Comment by d-exclaimation** [comment]
Right, is there any problem resuming after the delay or is it just that there's the delay before the connection is up again?

It's seemed like resumeConnection is just non-blocking to compensate for the WebSocket initiation time taking noticeably too much time

**Comment by d-exclaimation** [comment]
My testing doesn't show anything abnormal about the resuming HTTP upgrade request sent by Apollo iOS and nothing abnormal about how Pioneer respond to it as well. 

The delay in resuming was something I did notice, but it was mostly a fairly small delay <40ms roughly. I did notice even after the connection is fully resumed the subscription was still suspended for a brief moment, but I couldn't reliably reproduce that.
